sdks/core/src/test/java/org/gbif/pipelines/core/parsers/clustering/OccurrenceFeaturesPojo.java

```diff
@@ -1,75 +1,34 @@
 package org.gbif.pipelines.core.parsers.clustering;
 
-import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.Builder;
 
 /** A POJO implementation for simple tests. */
+@Builder
 public class OccurrenceFeaturesPojo implements OccurrenceFeatures {
-  private String id;
-  private String datasetKey;
-  private Integer speciesKey;
-  private Integer taxonKey;
-  private String basisOfRecord;
-  private Double decimalLatitude;
-  private Double decimalLongitude;
-  private Integer year;
-  private Integer month;
-  private Integer day;
-  private String eventDate;
-  private String scientificName;
-  private String countryCode;
-  private String typeStatus;
-  private String occurrenceID;
-  private String recordedBy;
-  private String fieldNumber;
-  private String recordNumber;
-  private String catalogNumber;
-  private String otherCatalogNumbers;
-
-  private OccurrenceFeaturesPojo(
-      String id,
-      String datasetKey,
-      Integer speciesKey,
-      Integer taxonKey,
-      String basisOfRecord,
-      Double decimalLatitude,
-      Double decimalLongitude,
-      Integer year,
-      Integer month,
-      Integer day,
-      String eventDate,
-      String scientificName,
-      String countryCode,
-      String typeStatus,
-      String occurrenceID,
-      String recordedBy,
-      String fieldNumber,
-      String recordNumber,
-      String catalogNumber,
-      String otherCatalogNumbers) {
-    this.id = id;
-    this.datasetKey = datasetKey;
-    this.speciesKey = speciesKey;
-    this.taxonKey = taxonKey;
-    this.basisOfRecord = basisOfRecord;
-    this.decimalLatitude = decimalLatitude;
-    this.decimalLongitude = decimalLongitude;
-    this.year = year;
-    this.month = month;
-    this.day = day;
-    this.eventDate = eventDate;
-    this.scientificName = scientificName;
-    this.countryCode = countryCode;
-    this.typeStatus = typeStatus;
-    this.occurrenceID = occurrenceID;
-    this.recordedBy = recordedBy;
-    this.fieldNumber = fieldNumber;
-    this.recordNumber = recordNumber;
-    this.catalogNumber = catalogNumber;
-    this.otherCatalogNumbers = otherCatalogNumbers;
-  }
+  private final String id;
+  private final String datasetKey;
+  private final Integer speciesKey;
+  private final Integer taxonKey;
```

Comment:
Sorry to ask but is a requirement that speciesKey and taxonKey be Integers ?

Reply:
After discussion with @timrobertson100 ive raised this issue: https://github.com/gbif/pipelines/issues/484